controllers/utils/events.go

```diff
@@ -32,8 +32,6 @@ func (c *ComplianceEventSender) SendEvent(
 	msg string,
 	compliance policyv1.ComplianceState,
 ) error {
-	msg = string(compliance) + "; " + msg
-
```

Comment:
I think removing it here will remove the status from ALL messages (ConfigurationPolicy, IamPolicy, CertificatePolicy, Gatekeeper) rather than just the duplicated Gatekeeper messages.

Reply:
Assuming that's the case and this is affecting all policies, the cause might be somewhere in the Gatekeeper sync code instead: https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/main/controllers/gatekeepersync/gatekeeper_constraint_sync.go

Reply:
From what I can gather, the cause seems to come from calls to the `emitTemplateEvent()` function in `template_sync.go`. The `msgMeta` field is supplied with the extra compliance text and concatenated with the compliance state in the subsequent call to `sendEvent()` in `events.go`.

Reply:
There is also a similar function in `gatekeeper_contraint_sync.go` called `sendComplianceEvent` (https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/main/controllers/gatekeepersync/gatekeeper_constraint_sync.go#L347) that wraps around `sendEvent`. In here, the `msg` field is not supplied with a compliance prefix.

Reply:
Ah, I see what you mean about the `SendEvent` function is appending the compliance state, so I'm thinking this isn't just limited to Gatekeeper policies and that's just a coincidence I noticed it there. 🤔 
It might be best to stick a check or two in an E2E to check these messages (if they're not checked already) since it feels like this code is arriving here from a few different places.
Also `emitTemplateError/Success/Pending` could instead pass `policiesv1.ComplianceState` directly instead of `msgMeta`, and `emitTemplateError` could add the `template-error; ` string to the `msg` instead of `msgMeta`?

Reply:
I was actually thinking about moving the `template-error; ` string to the `msg` field but wasn't sure if I should since it seemed like a piece of  "metadata". I'll go ahead and make those changes now.